lib/ansible/module_utils/network/common/facts/facts.py

```diff
@@ -20,7 +20,9 @@ def __init__(self, module):
         self._warnings = []
         self._gather_subset = module.params.get('gather_subset')
         self._gather_network_resources = module.params.get('gather_network_resources')
-        self._connection = get_resource_connection(module)
+        self._connection = None
+        if module.params['state'] not in ['rendered', 'parsed']:
```

Comment:
I think for scenario when we are only gathering facts for some RM , it will fail  right ?